core/lib/l1_contract_interface/src/i_executor/commit/trusted_setup.rs

```diff
@@ -0,0 +1,4176 @@
+use std::{convert::TryInto, iter};
+
+use const_decoder::Decoder::Hex;
+use once_cell::sync::Lazy;
+use zkevm_test_harness_1_4_2::{
+    boojum::pairing::{bls12_381::G2Compressed, EncodedPoint},
+    kzg::KzgSettings,
+    zkevm_circuits::{
+        boojum::pairing::{
+            bls12_381::{Fr, FrRepr, G1Compressed},
+            ff::{Field as _, PrimeField as _},
+            CurveAffine,
+        },
+        eip_4844::input::ELEMENTS_PER_4844_BLOCK,
+    },
+};
+
+const G2_POINT_1_BYTES: [u8; 96] = Hex.decode(
+    b"b5bfd7dd8cdeb128843bc287230af38926187075cbfbefa81009a2ce615ac53d2914e5870cb452d2afaaab24f3499f72\
+      185cbfee53492714734429b7b38608e23926c911cceceac9a36851477ba4c60b087041de621000edc98edada20c1def2"
+);
+
+const FIRST_ROOT_OF_UNITY: FrRepr = FrRepr([
+    0xe206da11a5d36306,
+    0x0ad1347b378fbf96,
+    0xfc3e8acfe0f8245f,
+    0x564c0a11a0f704f4,
+]);
+
+fn bit_reverse_slice_indices<T>(array: &mut [T]) {
+    assert_eq!(array.len(), ELEMENTS_PER_4844_BLOCK);
+    for idx in 0..ELEMENTS_PER_4844_BLOCK {
+        let reversed_idx = idx.reverse_bits() >> (usize::BITS - ELEMENTS_PER_4844_BLOCK.ilog2());
+        if idx < reversed_idx {
+            array.swap(idx, reversed_idx);
+        }
+    }
+}
+
+pub(super) static KZG_SETTINGS: Lazy<KzgSettings> = Lazy::new(|| {
+    let first_root_of_unity =
+        Fr::from_repr(FIRST_ROOT_OF_UNITY).expect("invalid first root of unity");
+    let mut roots_of_unity: Box<[_]> = iter::successors(Some(Fr::one()), |prev| {
+        let mut next = first_root_of_unity;
+        next.mul_assign(prev);
+        Some(next)
+    })
+    .take(ELEMENTS_PER_4844_BLOCK)
+    .collect();
+    bit_reverse_slice_indices(&mut roots_of_unity);
+
+    let mut setup_g2_1 = G2Compressed::empty();
+    setup_g2_1.as_mut().copy_from_slice(&G2_POINT_1_BYTES);
+    let setup_g2_1 = setup_g2_1
+        .into_affine()
+        .expect("invalid G2 point #1")
+        .into_projective();
+
+    let mut lagrange_setup: Box<[_]> = L1_LAGRANGE
+        .iter()
+        .map(|g1_bytes| {
+            let mut g1 = G1Compressed::empty();
+            g1.as_mut().copy_from_slice(g1_bytes);
+            g1.into_affine().expect("invalid G1 point")
+        })
+        .collect();
+    bit_reverse_slice_indices(&mut lagrange_setup);
+
+    KzgSettings {
+        roots_of_unity_brp: roots_of_unity.try_into().unwrap(),
+        setup_g2_1,
+        lagrange_setup_brp: lagrange_setup.try_into().unwrap(),
+    }
+});
+
+// Taken from the C KZG library: https://github.com/ethereum/c-kzg-4844/blob/main/src/trusted_setup.txt
+// with added `Hex.decode(_)` wrappers.
+#[allow(long_running_const_eval)]
+const L1_LAGRANGE: [[u8; 48]; ELEMENTS_PER_4844_BLOCK] = [
```

Comment:
There are a couple of other ways to include these constants, e.g. parse them using a build script. I've chosen this approach because it's more transparent and requires less privileges during the build.

Reply:
Not a strong opinion, but given that it's already wrapped into `once_cell::Lazy` (especially with `L1_LAGRANGE.iter()`, maybe we'll have a `trusted_setup.txt` as a separate file and use `include_str!` here?

Reply:
Done. This means ~2x more space occupied in the binary, but I guess it's not the largest concern, and we could switch to using a build script later if this becomes an issue.